azurelinuxagent/ga/exthandlers.py

```diff
@@ -1399,6 +1439,43 @@ def get_seq_no(self):
         # Will have to revisit once the feature to enable multiple runtime settings is rolled out by CRP
         return self.ext_handler.properties.extensions[0].sequenceNumber
 
+    def _read_and_parse_json_status_file(self, ext, seq_no, ext_status_file):
+        failed_to_read = False
+        failed_to_parse_json = False
+        raised_exception = None
+        data_str = None
+        data = None
+
+        for attempt in range(_NUM_OF_STATUS_FILE_RETRIES):
+            try:
+                data_str = fileutil.read_file(ext_status_file)
+                data = json.loads(data_str)
+                break
+            except IOError as e:
+                failed_to_read = True
+                raised_exception = e
+            except ValueError as e:
+                failed_to_parse_json = True
+                raised_exception = e
+            time.sleep(_STATUS_FILE_RETRY_DELAY)
+
+        if failed_to_read:
+            msg = u"We couldn't find any status for {0}-{1} extension, for the sequence number {2}. It failed due to " \
```

Comment:
we talked about changing this from "couldn't find" to "couldn't read"

Reply:
also 'StatusFileNotFound' is misleading

Reply:
Done.